src/_pytest/pytester.py

```diff
@@ -750,11 +750,16 @@ def make_hook_recorder(self, pluginmanager: PytestPluginManager) -> HookRecorder
         self._request.addfinalizer(reprec.finish_recording)
         return reprec
 
-    def chdir(self) -> None:
+    def chdir(self, new_path: Optional[Path] = None) -> None:
         """Cd into the temporary directory.
 
         This is done automatically upon instantiation.
+
+        :param new_path:
+            Optionally specify a different path to cd into
         """
+        if new_path:
+            self._path = new_path
```

Comment:
I don't think we should change `self._path`, only change to the new directory:

```python
chdir_to = new_path if new_path is not None else self._path
self._monkeypatch.chdir(chdir_to)

```

Changing `_path` changes the original location of the test directory, which is incorrect.

Reply:
oh, the test i was doing this for was checking `pytester.path` later, but i see that's not actually needed and after removing that check and instead making it a relative path, i was able to just use `monkeypatch` normally as suggested in https://github.com/pytest-dev/pytest/issues/11942#issuecomment-1935584253

so i guess this PR isn't really needed